src/utils/advance.js

```diff
@@ -0,0 +1,42 @@
+import { Random } from "@woowacourse/mission-utils";
+import { GAME_RULES } from "../constants/index.js";
+
+// 전진 조건 체크
+export const shouldAdvance = (randomNumber) => {
+  return randomNumber >= GAME_RULES.ADVANCE_NUMBER;
+};
+
+// 자동차 전진 로직(전진 카운트 증가)
+export const advanceCars = (carNames, incrementAdvanceCount) => {
+  const { MIN_RANDOM_NUMBER, MAX_RANDOM_NUMBER } = GAME_RULES;
+  carNames.forEach((carName) => {
+    const randomNumber = Random.pickNumberInRange(MIN_RANDOM_NUMBER, MAX_RANDOM_NUMBER);
+
+    if (shouldAdvance(randomNumber)) {
+      incrementAdvanceCount(carName);
+    }
+  });
+};
+
+// 각 자동차의 전진 카운트를 대시로 변환해서 결과 리턴
+export const getAdvanceResult = (carsInfoEntries) => {
+  const result = carsInfoEntries
+    .map(([carName, advanceCount]) => {
+      const dash = GAME_RULES.DASH.repeat(advanceCount);
+      return `${carName} : ${dash}`;
+    })
+    .join("\n");
+
+  return result;
+};
```

Comment:
이 부분에서도 중간 변수를 제거하는 편이 조금 더 간결해 질 수 있을 것 같아요!
```suggestion
export const getAdvanceResult = (carsInfoEntries) =>
  carsInfoEntries
    .map(([carName, advanceCount]) => `${carName} : ${GAME_RULES.DASH.repeat(advanceCount)}`)
    .join("\n");
```

Reply:
저는 전자가 가독성이 더 좋다고 생각했던 것 같아요 🤔